### PR TITLE
w3ip-02 test: withdraw code staking & suicide

### LINKED ISSUE #23: w3ip-2 tests: removing stakes and selfdestruct
we need to cover 2 more test cases:

1. when a contract has already stored a huge amount of code with sufficient stakings, the stakes can't be withdrawn
2. once the above-mentioned contract is destroyed, its stakings should be sent to the caller 

a simple contract will be used in such tests

### PR DESCRIPTION
**Pull request by ninjaahhh**
fixes #23

### CODE REVIEW

File: core/vm/evm_test.go [hunk 1]
```diff
@@ -148,44 +157,109 @@ var withdrawStakingTests = []struct {
...
-		statedb.CreateAccount(addr)
-		statedb.SetCode(addr, code)
-		statedb.SetBalance(addr, initBal)
+	calls := []string{"call", "callCode", "delegateCall"}
+	// contract Contract {
+	//   function withdraw(uint256 amount, address payable to) external payable {
+	//     to.transfer(amount);
+	//   }
+	// }
+	initCode := hexutil.MustDecode("0x60806040526004361061001d5760003560e01c8062f714ce14610022575b600080fd5b61003c60048036038101906100379190610122565b61003e565b005b8073ffffffffffffffffffffffffffffffffffffffff166108fc839081150290604051600060405180830381858888f19350505050158015610084573d6000803e3d6000fd5b505050565b600080fd5b6000819050919050565b6100a18161008e565b81146100ac57600080fd5b50565b6000813590506100be81610098565b92915050565b600073ffffffffffffffffffffffffffffffffffffffff82169050919050565b60006100ef826100c4565b9050919050565b6100ff816100e4565b811461010a57600080fd5b50565b60008135905061011c816100f6565b92915050565b6000806040838503121561013957610138610089565b5b6000610147858286016100af565b92505060206101588582860161010d565b915050925092905056fea264697066735822122087ac4dd4a397d6abb35fd02d3945c392429ab58f1bcf76e724e6e8534373e84d64736f6c634300080c0033")
```
Comment (qizhou) [inline core/vm/evm_test.go:166]: Any reference to find the source of the code?

Comment (ninjaahhh) [reply core/vm/evm_test.go:166]: remix compiler tab "Compilation Details" -> "Runtime Bytecode" 
image

Comment (qizhou) [reply core/vm/evm_test.go:166]: I know, but I mean the **source code** and its compilation parameters.

Comment (ninjaahhh) [reply core/vm/evm_test.go:166]: source code is right there between line 161 to 165.
will probably add compiler version to the comments as well.

**Review by qizhou — APPROVED** [review]

**Pull request merged**